compiler/resolution/preFold.cpp

```diff
@@ -458,6 +460,140 @@ static void setRecordDefaultValueFlags(AggregateType* at) {
   }
 }
 
+//
+// Infer the correct storage kind for a context manager resource when no
+// explicit storage is given, e.g.
+//
+//    manage myManager() as myResource do ...;
+//
+// Look for PRIM_INIT_VAR calls where the LHS is a variable marked with
+// FLAG_MANAGER_RESOURCE_INFER_STORAGE, then chase the call temp on
+// the RHS (it should always be a call temp).
+//
+// If the call temp was initialized by a ContextCall, search through the CC
+// and record if:
```

Comment:
Yeah, this is pretty much what cull over references is supposed to do. I think we should move this code to cullOverReferences. But, given that we are porting over resolution to the new compiler, and this is arguably a workaround for #18205, I don't see moving it as particularly urgent (i.e. I'm OK with it being this way for the release). However I would ask that at the very least in this PR you add a TODO comment here that this should really be happening in cullOverReferences.

Reply:
Thanks, I'll add a TODO.